src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerInEstablishmentResource.kt

```diff
@@ -0,0 +1,82 @@
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.resource.advice.ErrorResponse
+import uk.gov.justice.digital.hmpps.prisonersearch.services.PrisonerInEstablishmentService
+import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.PrisonerInEstablishmentRequest
+import javax.validation.Valid
+
+@RestController
+@Validated
+@PreAuthorize("hasAnyRole('ROLE_GLOBAL_SEARCH', 'ROLE_PRISONER_SEARCH')")
+class PrisonerInEstablishmentResource(private val searchService: PrisonerInEstablishmentService) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  // A hack to allow swagger to determine the response schema which includes generic content
+  abstract class Response : Page<Prisoner>
+
+  @Operation(
+    summary = "Search for a prisoner within a particular establishment",
+    description = """ 
+      This search is optimised for clients that have a simple search term typically containing the prisonser's name
+      or prisoner number. The user typically is certain the prisoner is within the establishment and knows key information 
+      about the prisoner.
+      Requires ROLE_GLOBAL_SEARCH or ROLE_PRISONER_SEARCH role.
+      """,
+    security = [SecurityRequirement(name = "ROLE_GLOBAL_SEARCH"), SecurityRequirement(name = "ROLE_PRISONER_SEARCH")],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = PrisonerInEstablishmentRequest::class)
+        )
+      ]
+    ),
+
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Search successfully performed",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = Response::class))]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect information provided to perform prisoner match",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to search for prisoner data",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+    ]
+  )
+  @PostMapping("/establishment/{prisonId}/prisoner", produces = [MediaType.APPLICATION_JSON_VALUE])
```

Comment:
I find this terminology a bit odd.  Why call the prison an establishment when the parameter is then `prisonId` rather than `establishmentId`?  Do you need the `/prisoner` on the end too given that this the `prisoner-offender-search` (which I think should be renamed to just `prisoner-search` or `prison-offender-search` or I guess more correctly `prison-prisoner-search` since offender is being replaced by prisoner...

Reply:
Maybe; in NOMIS it is called "establishment" search. DPS it has no name really.
I need a name that makes it clear it isn't searching for an establishment but in an establishment, hence the /prisoner part - otherwise you end up with `/establishment/MDI` which feels like you are "finding" Moorland not prisoners in Moorland.

Maybe this 

`/prison/MDI/prisoner`  ?